CatAnalyzer/plugins/dileptonCommon.cc

```diff
@@ -294,84 +300,116 @@ int dileptonCommon::eventSelection(const edm::Event& iEvent, const edm::EventSet
       }
     }
 
-    edm::Handle<vector<int> > partonTop_modes;
-    edm::Handle<reco::GenParticleCollection> partonTop_genParticles;
-    iEvent.getByToken(partonTop_modes_, partonTop_modes);
-    iEvent.getByToken(partonTop_genParticles_, partonTop_genParticles);
-    if ( (*partonTop_modes).size() == 0 ) {
-      b_gen_partonMode1 = 0;
-      b_gen_partonMode2 = 0;
-    }
-    else if ( (*partonTop_modes).size() == 1 ) { b_gen_partonMode2 = 0; }
-    else{
-      b_gen_partonChannel = *partonTop_channel;
-      b_gen_partonMode1 = (*partonTop_modes)[0];
-      b_gen_partonMode2 = (*partonTop_modes)[1];
+    edm::Handle<vector<reco::GenParticle> > genParticleHandle;
+    iEvent.getByToken(partonTop_, genParticleHandle);
```

Comment:
Looks like you are redoing the partonTopProducer. Is there any problem to use it? Maybe you can update the PartonTopProducer, not here if you observe any problem or suggestion to improve it.